src/components/button/button.ts

```diff
@@ -233,18 +256,45 @@ export class Button extends LitElement {
 
   /** @internal */
   private _debounceResize = debounce(() => {
+    this._handleScroll();
     this.iconOnly = this._testIconOnly();
   });
 
+  /** @internal */
+  private reSizeButton() {
+    // Resize button to medium at mobile breakpoint
+    this.reSizeBtn =
+      (this.isFloating || this.showOnScroll) && window.innerWidth <= 672
+        ? true
+        : false;
+  }
+
+  /** @internal */
+  private _handleScroll() {
+    if (this.showOnScroll) {
+      const scrollPosition = window.scrollY;
+      const windowHeight = window.innerHeight;
+      const pageHeight = document.documentElement.scrollHeight;
+      // Show the button if scrolled past halfway
+      this.showButton = scrollPosition > (pageHeight - windowHeight) / 2;
+    } else {
+      this.showButton = true; // Always visible
+    }
+  }
+
   override connectedCallback() {
     super.connectedCallback();
-
     window.addEventListener('resize', this._debounceResize);
+    if (this.showOnScroll) {
+      window.addEventListener('scroll', this._handleScroll.bind(this));
```

Comment:
I would suggest feeding this though the debounce function similar to the resize listener. That way it only executes after a batch of scrolling, rather than on every pixel scrolled.

Reply:
Addressed.